### PR TITLE
Replaced compile-time flag IS_TEST_NET with a dyn configurable one

### PR DESCRIPTION
**Pull request by Katkevich**

### CODE REVIEW

File: libraries/chain/database/database.cpp [hunk 1]
```diff
@@ -875,8 +875,9 @@ void database::notify_post_apply_operation(const operation_notification& note)
 
 inline void database::push_virtual_operation(const operation& op)
 {
-#if defined(IS_LOW_MEM) && !defined(IS_TEST_NET)
-    return;
+#if defined(IS_LOW_MEM)
```
Comment (alexplevako) [inline libraries/chain/database/database.cpp]: try remove this check and run tests.

File: libraries/chain/evaluators/scorum_evaluators.cpp [hunk 2]
```diff
@@ -189,12 +189,14 @@ void account_update_evaluator::do_apply(const account_update_operation& o)
 
     if (o.owner)
     {
-#ifndef IS_TEST_NET
-        dynamic_global_property_service_i& dprops_service = db().dynamic_global_property_service();
+        if (!protocol::detail::config::is_test_net)
+        {
+            dynamic_global_property_service_i& dprops_service = db().dynamic_global_property_service();
+
+            FC_ASSERT(dprops_service.head_block_time() - account_auth.last_owner_update > SCORUM_OWNER_UPDATE_LIMIT,
```
Comment (alexplevako) [inline libraries/chain/evaluators/scorum_evaluators.cpp]: decrease 'SCORUM_OWNER_UPDATE_LIMIT' and remove parent `if`

File: libraries/chain/evaluators/scorum_evaluators.cpp [hunk 3]
```diff
@@ -836,9 +838,8 @@ void vote_evaluator::do_apply(const vote_operation& o)
 
         int64_t elapsed_seconds = (dprops_service.head_block_time() - voter.last_vote_time).to_seconds();
 
-#ifndef IS_TEST_NET
-        FC_ASSERT(elapsed_seconds >= SCORUM_MIN_VOTE_INTERVAL_SEC, "Can only vote once every 3 seconds.");
-#endif
+        if (!protocol::detail::config::is_test_net)
+            FC_ASSERT(elapsed_seconds >= SCORUM_MIN_VOTE_INTERVAL_SEC, "Can only vote once every 3 seconds.");
```
Comment (alexplevako) [inline libraries/chain/evaluators/scorum_evaluators.cpp]: decrease `SCORUM_MIN_VOTE_INTERVAL_SEC` and remove parent if

File: libraries/chain/include/scorum/chain/database/database.hpp [hunk 4]
```diff
@@ -49,16 +49,22 @@ class database : public chainbase::database,
...
 
 public:
+    enum creation_option
+    {
+        opt_none = 0,
+        opt_log_hardforks = 1 << 0,
+        opt_notify_virtual_op_applying = 1 << 1
+    };
+
     database();
```
Comment (alexplevako) [inline libraries/chain/include/scorum/chain/database/database.hpp]: delete default constructor if it is not used.

File: Dockerfile [hunk 5]
```diff
@@ -51,37 +51,13 @@ RUN \
...
-    make -j$(nproc) && \
-    ./libraries/chainbase/test/chainbase_test && \
-    ./tests/utests/utests && \
-    ./tests/chain_tests/chain_tests && \
-    ./tests/wallet_tests/wallet_tests && \
-    ./programs/util/test_fixed_string && \
-    cd /usr/local/src/scorum && \
-    doxygen && \
-    programs/build_helpers/check_reflect.py && \
-    programs/build_helpers/get_config_check.sh && \
```
Comment (Andrei-Masilevich) [inline Dockerfile]: move all these checks:

    doxygen && \
    programs/build_helpers/check_reflect.py && \
    programs/build_helpers/get_config_check.sh && \

 in to uncommented section. Check result: 

sudo docker build . -t test

File: libraries/chain/include/scorum/chain/database/database.hpp [hunk 6]
```diff
@@ -49,16 +49,27 @@ class database : public chainbase::database,
 {
 
 public:
-    database();
+    enum creation_option
+    {
+        opt_none = 0,
+        opt_log_hardforks = 1 << 0,
```
Comment (Andrei-Masilevich) [inline libraries/chain/include/scorum/chain/database/database.hpp]: Overhead. We all know degrees of two (1, 2).
Replace opt_log_hardforks by 1 and opt_notify_virtual_op_applying by 2

**Pull request merged**